hivemind/client/averaging/__init__.py

```diff
@@ -348,8 +348,16 @@ def _background_thread_fetch_current_state_if_asked(self):
                 logger.warning(e)
                 continue
 
-    def load_state_from_peers(self, wait=True) -> Optional[Any]:
-        """ Try to download the latest optimizer state one of the existing peer """
+    def load_state_from_peers(self, wait=True) -> Optional[Tuple[bytes, Sequence[torch.Tensor]]]:
+        """
+        Try to download the latest optimizer state one of the existing peer.
+        :returns: on success, return a 2-tuple with (serialized_metadata, tensors), where
+
+        - serialized_metadata is a small bytestring containing **serialized** metadata (e.g. hyperparameters)
```

Comment:
Shouldn't we also deserialize on return? What are the cases where we still need serialized metadata as the output of this method?

Reply:
As of now, this is left to the user (to provide choose a secure pickler).